Make terraform sometimes fails with ResourcesNotFoundException

*Note: this issue will exist in dev after #556 is resolved*

Here is the error [as encountered](https://github.com/DataBiosphere/azul/pull/1635#issuecomment-608157775) by @hannes-ucsc:
```
module.chalice_indexer.aws_lambda_function.write: Creation complete after 3m22s [id=azul-indexer-sandbox-write]

Error: Error adding new Lambda Permission for azul-indexer-sandbox-indexercachehealth: ResourceNotFoundException: Function not found: arn:aws:lambda:us-east-1:122796619775:function:azul-indexer-sandbox-indexercachehealth
{
  Message_: "Function not found: arn:aws:lambda:us-east-1:122796619775:function:azul-indexer-sandbox-indexercachehealth",
  Type: "User"
}

  on indexer/chalice.tf.json line 297, in resource.aws_lambda_permission.indexercachehealth-event:
 297:             },



Error: Error adding new Lambda Permission for azul-indexer-sandbox-nudge: ResourceNotFoundException: Function not found: arn:aws:lambda:us-east-1:122796619775:function:azul-indexer-sandbox-nudge
{
  Message_: "Function not found: arn:aws:lambda:us-east-1:122796619775:function:azul-indexer-sandbox-nudge",
  Type: "User"
}

  on indexer/chalice.tf.json line 303, in resource.aws_lambda_permission.nudge-event:
 303:             },



Error: Error adding new Lambda Permission for azul-service-sandbox-servicecachehealth: ResourceNotFoundException: Function not found: arn:aws:lambda:us-east-1:122796619775:function:azul-service-sandbox-servicecachehealth
{
  Message_: "Function not found: arn:aws:lambda:us-east-1:122796619775:function:azul-service-sandbox-servicecachehealth",
  Type: "User"
}

  on service/chalice.tf.json line 337, in resource.aws_lambda_permission.servicecachehealth-event:
 337:             },


make[1]: *** [Makefile:18: apply] Error 1
make[1]: Leaving directory '/home/hannes/workspace/hca/azul/terraform'
make: *** [Makefile:9: terraform] Error 2
```

The workaround is to simply rerun `make terraform`.

Solutions include:
- patching the configuration generated by Chalice so that `aws_lambda_permission.*.function_name` is a terraform reference instead of a direct string.
- fixing Chalice so that the generated code doesn't need to be patched.

--- a/chalice/package.py
+++ b/chalice/package.py
@@ -925,7 +925,7 @@ class TerraformGenerator(TemplateGenerator):
         template['resource'].setdefault(
             'aws_lambda_permission', {})[
             resource.resource_name] = {
-            'function_name': resource.lambda_function.function_name,
+            'function_name': self._fref(resource.lambda_function, attr='function_name'),
             'action': 'lambda:InvokeFunction',
             'principal': self._options.service_principal('events'),
             'source_arn': "${aws_cloudwatch_event_rule.%s.arn}" % (

--- a/setup.py
+++ b/setup.py
@@ -23,7 +23,7 @@ install_requires = [
 
 setup(
     name='chalice',
-    version='1.18.0+5',
+    version='1.18.0+6',
     description="Microframework",
     long_description=README,
     author="James Saryerwinnie",

